dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,345 @@
+decentralized process.
+
+The main changes are:
+
+* Move multisig utxo based treasury address to a treasury account that can only be debited when Decred stakeholders agree to it.
+* Credit development subsidy to treasury account every block.
+* To facilitate the changes three new opcodes <code>OP_TADD</code> <code>OP_TSPEND</code> and <code>OP_TGEN</code> were added.
+* The aforementioned opcodes are then in turn used to construct the three new types of stake transactions:
+*# Treasurybase transaction - treasury subsidy in every block
+*# Treasury add transactions - method to credit the treasury account.
+*# Treasury spend transactions - method to debit the treasury account.
+
+==Motivation==
+
+At this present time all Decred treasury payouts are handled by DHG. This is
+undesirable for various reasons, such as
+
+* sovereignty over treasury spends is not vested in the stakeholders
+* risk of theft of funds
+* risk of key project members abandoning the project
+* risk of coercion being used against key project members
+* process of treasury spending is insufficiently transparent
+
+The proposed solution will semi-automate the release of treasury funds while
+diminishing key project members' influence over the treasury spends.
+
+==Process==
+
+Currently, every block that is mined must include a coinbase transaction that
+that generates a utxo of a specific amount that requires a 2-of-3 multisig to
+spend. This is enforced by consensus. During payout time, DHG creates and signs
+transactions to all payees. This process is not adequate to support the
+features proposed in this DCP.
+
+Essentially, the treasury is converted from a utxo based model to an account
+based model. The account balance is determined by an initial state followed by
+state changes that are recorded on the blockchain. This process "burns" and
+"creates" utxos as needed. Sending utxos to the treasury burns the coins and
+adds value to the treasury account and sending from the treasury creates utxos.
+This process is analogous to the ticket purchases and redeems. The treasury
+changes shall be added to the staking mechanism.
+
+This DCP introduces three distinct new processes.
+# In order to credit the treasury account every block the coinbase transaction is split into two transactions. In the stake tree the first transaction is now a treasurybase. Treasurybase is a newly introduced term that is defined as the transaction that credits the treasury account on every block. The value of this transaction is the same (exception will be specified later in this document) as it was in the coinbase transaction. The coinbase transaction remains identical except for the noted change.
+# Users need to be able to send utxos to the treasury account. This is a coin burn that is analogous to purchasing a ticket. This feature is also going to be used to move currently unused funds in the treasury to the new treasury account.
```

Comment:
This is a specification.  I don't think there is any reason to call out that the feature will be used to move over the current treasury funds here.

```suggestion
# Crediting the Treasury from existing utxos is handled by a new <code>treasury add</code> stake transaction type that burns the utxo(s) and credits the Treasury account accordingly.  This is analogous to purchasing a ticket.
```